### ISSUE TITLE
'else if constexpr' fails with readability/braces

### ISSUE THREAD

**Issue by Shachlan** [cpplint/cpplint#135]
"If/else bodies with multiple statements require braces  [readability/braces] " error is given for code like

```cpp
if constexpr (foo) {
...
} else if constexpr (bar) {
...
} else {
...
}
```

while 

```cpp
if (foo) {
...
} else if (bar) {
...
} else {
...
}
```

is fine.

**Comment by tkruse** [comment]
This could additionally be reported upstream at https://github.com/cpplint/cpplint/issues/135

PRs are welcome to fix this.

**Comment by Shachlan** [comment]
https://github.com/google/styleguide/issues/541
https://github.com/cpplint/cpplint/pull/136
